mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -58,6 +59,61 @@ struct MakerRefundSecretArgs<'a> {
 }
 
 impl EthCoin {
+    pub(crate) async fn get_fee_to_send_maker_payment_v2_impl(
+        &self,
+        args: GetFeeToSendMakerPaymentArgs,
+    ) -> TradePreimageResult<TradeFee> {
+        const NON_ZERO_SECRET_HASH: [u8; 32] = [1_u8; 32];
+        const NON_EMPTY_ADDRESS: [u8; 20] = [1_u8; 20];
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .ok_or_else(|| {
+                TradePreimageError::InternalError(ERRL!("Expected swap_v2_contracts to be Some, but found None"))
+            })?
+            .maker_swap_v2_contract;
+        let payment_amount = u256_from_big_decimal(&args.amount, self.decimals)
+            .map_err(|err| TradePreimageError::InternalError(ERRL!("{}", err)))?;
+        let payment_args = {
+            MakerPaymentArgs {
+                taker_address: Address::from_slice(&NON_EMPTY_ADDRESS),
+                taker_secret_hash: &NON_ZERO_SECRET_HASH,
+                maker_secret_hash: &NON_ZERO_SECRET_HASH,
+                payment_time_lock: args.time_lock,
+            }
+        };
+        let (eth_value, data) = match &self.coin_type {
+            EthCoinType::Eth => {
+                let data = self
+                    .prepare_maker_eth_payment_data(&payment_args)
+                    .await
+                    .map_err(|err| TradePreimageError::InternalError(ERRL!("{}", err)))?;
+                (payment_amount, data)
+            },
+            EthCoinType::Erc20 {
+                platform: _,
+                token_addr,
+            } => {
+                let data = self
+                    .prepare_maker_erc20_payment_data(&payment_args, payment_amount, *token_addr)
+                    .await
+                    .map_err(|err| TradePreimageError::InternalError(ERRL!("{}", err)))?;
+                (0.into(), data)
+            },
+            EthCoinType::Nft { .. } => return MmError::err(TradePreimageError::NftProtocolNotSupported),
+        };
+        let (gas_limit, fee_per_gas) = self
+            .estimate_gas_for_contract_call_if_conf(maker_swap_v2_contract, Bytes::from(data.clone()), eth_value)
```

Comment:
Here we create fake swap params for estimate gas for maker payment.
Another option is just to use the gas_limit constant (like this is done for other calls, e.g get_fee_to_send_taker_funding_impl) - this looks more reliable but implies more gas fee shown in the trade_preimage RPC.